modmail/extensions/extension_manager.py

```diff
@@ -0,0 +1,276 @@
+
+    The * and ** values bypass this when used with the reload command.
+    """
+
+    source_list = EXTENSIONS
+    type = "extension"
+
+    async def convert(self, ctx: Context, argument: str) -> str:
+        """Fully qualify the name of an extension and ensure it exists."""
+        # Special values to reload all extensions
+        if argument == "*" or argument == "**":
+            return argument
+
+        argument = argument.lower()
+
+        if argument in self.source_list:
+            return argument
+
+        qualified_arg = f"modmail.{self.type}s.{argument}"
+        if qualified_arg in self.source_list:
+            return qualified_arg
+
+        matches = []
+        for ext in self.source_list:
+            if argument == unqualify(ext):
+                matches.append(ext)
+
+        if not matches:
+            raise commands.BadArgument(f":x: Could not find the {self.type} `{argument}`.")
+
+        if len(matches) > 1:
+            names = "\n".join(sorted(matches))
+            raise commands.BadArgument(
+                f":x: `{argument}` is an ambiguous {self.type} name. "
+                f"Please use one of the following fully-qualified names.```\n{names}```"
+            )
+
+        return matches[0]
+
+
+class ExtensionManager(ModmailCog):
+    """Extension management base class."""
+
+    type = "extension"
+
+    def __init__(self, bot: ModmailBot):
+        self.bot = bot
+        self.all_extensions = EXTENSIONS
+
+    async def get_black_listed_extensions() -> list:
+        """Returns a list of all blacklisted extensions."""
+        raise NotImplementedError()
+
+    @commands.group("ext", aliases=("extensions", "exts"), invoke_without_command=True)
+    async def extensions_group(self, ctx: Context) -> None:
+        """Load, unload, reload, and list loaded extensions."""
+        await ctx.send_help(ctx.command)
+
+    @extensions_group.command(name="load", aliases=("l",))
+    async def load_extensions(self, ctx: Context, *extensions: ExtensionConverter) -> None:
+        """
+        Load extensions given their fully qualified or unqualified names.
+
+        If '\*' or '\*\*' is given as the name, all unloaded extensions will be loaded.
+        """  # noqa: W605
+        if not extensions:
+            await ctx.send_help(ctx.command)
+            return
+
+        if "*" in extensions or "**" in extensions:
+            extensions = sorted(ext for ext in self.all_extensions if ext not in self.bot.extensions.keys())
+
+        msg = self.batch_manage(Action.LOAD, *extensions)
+        await ctx.send(msg)
+
+    @extensions_group.command(name="unload", aliases=("ul",))
+    async def unload_extensions(self, ctx: Context, *extensions: ExtensionConverter) -> None:
+        """
+        Unload currently loaded extensions given their fully qualified or unqualified names.
+
+        If '\*' or '\*\*' is given as the name, all loaded extensions will be unloaded.
+        """  # noqa: W605
```

Comment:
Not needed.

```suggestion
        """
        Unload currently loaded extensions given their fully qualified or unqualified names.

        If '*' or '**' is given as the name, all loaded extensions will be unloaded.
        """
```

Reply:
Due to discord markdown, this is needed.